### ISSUE TITLE
Generic pre-deploy checks

### ISSUE THREAD

**Issue by JesperDramsch** [getnikola/nikola#3695]
**Requested Feature:** A generic hook into the deploy action to check a list of user-supplied criteria.

**Related Area:** Tasks? Could be an extra task to run.

**Do you want to contribute this yourself as a pull request?** (don’t worry about it if you don’t want to/can’t — someone else can take care of it)

- [ ] Yes, I have already written code for it (link if available and feasible)
- [x] Yes, I don’t have code ready yet (that’s okay!)
- [x] No (that’s okay too!)

**Does this feature affect backwards compatibility? If yes, in what way?**
No

**Rationale and full description:** (why should it be added to Nikola?)
Just like `nikola check` it would be fantastic to have a way to generically hook into the deploy action and check if a blog post meets user-defined criteria.

Basically, I always forget to add the `<!-- TEASER -->` thingy. And I would like the deploy to fail if my blog posts don't meet certain criteria, like containing that string. 

But I think having a list of criteria would just be really useful.

**Comment by Kwpolska** [comment]
Without requiring any changes being made to Nikola itself, you can leverage the fact that `nikola deploy` aborts if any command returns a non-zero exit code. You can write a simple Bash/Python script that scans your files in any way you want.
